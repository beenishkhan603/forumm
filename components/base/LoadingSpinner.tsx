import { useTheme } from '@libs/useTheme'
import Box from './Box'
import ForumLogoText from '@public/images/ForumLogoText.svg'
import ForumLogoRotating from '@public/images/ForumLogoRotating.svg'


const LoadingSpinner = ({
  className,
  size = 'small',
}: {
  size?: 'small' | 'medium' | 'large'
  className?: string
}) => {
  const { theme } = useTheme()

  const sizeClasses = {
    small: 'w-12 h-12',
    medium: 'w-24 h-24',
    large: 'w-32 h-32',
  }

  return (
    <Box
      data-testid="loading"
      className={`${sizeClasses[size]} ${className ?? ''}`}
      style={{ color: theme.textColour }}
    >
      <ForumLogoRotating fill={theme.textColour} className={`${sizeClasses[size]} ${className ?? ''}` } />
      { (size !== 'small') && (<ForumLogoText fill={theme.textColour} className={`${sizeClasses[size]} ${className ?? ''} !h-12` } />)}
    </Box>
  )
}

export default LoadingSpinner
