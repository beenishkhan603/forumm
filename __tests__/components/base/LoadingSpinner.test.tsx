import LoadingSpinner from '@components/base/LoadingSpinner'
import { render, waitFor, screen } from '@testing-library/react'

it('renders profile image unchanged', () => {
  const { container } = render(<LoadingSpinner />)
  expect(container).toMatchSnapshot()
})

it('renders profile image unchanged', () => {
  render(<LoadingSpinner />)
  waitFor(async () => {
    const spinnerEl = await screen.findByTestId('loading')
    expect(spinnerEl).toBeInTheDocument()
    expect(spinnerEl).toHaveClass('w-4 h-4')
  })
})

it('renders profile image unchanged', () => {
  render(<LoadingSpinner size={'small'} />)
  waitFor(async () => {
    const spinnerEl = await screen.findByTestId('loading')
    expect(spinnerEl).toBeInTheDocument()
    expect(spinnerEl).toHaveClass('w-4 h-4')
  })
})

it('renders profile image unchanged', () => {
  render(<LoadingSpinner size={'medium'} />)
  waitFor(async () => {
    const spinnerEl = await screen.findByTestId('loading')
    expect(spinnerEl).toBeInTheDocument()
    expect(spinnerEl).toHaveClass('w-12 h-12')
  })
})

it('renders profile image unchanged', () => {
  render(<LoadingSpinner size={'large'} />)
  waitFor(async () => {
    const spinnerEl = await screen.findByTestId('loading')
    expect(spinnerEl).toBeInTheDocument()
    expect(spinnerEl).toHaveClass('w-32 h-32')
  })
})
