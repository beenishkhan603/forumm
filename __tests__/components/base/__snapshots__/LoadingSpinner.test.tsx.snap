// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile image unchanged 1`] = `
<div>
  <div
    class="w-12 h-12 "
    data-testid="loading"
    style="color: rgb(24, 26, 32);"
  >
    <svg
      class="w-12 h-12 "
    />
  </div>
</div>
`;
